pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -436,3 +456,102 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
 	assert.Equal(t, 2, controller.GetAddressGroupNum())
 	assert.Equal(t, 1, controller.GetAppliedToGroupNum())
 }
+
+func TestNetworkPolicyMetrics(t *testing.T) {
+	// Initialize NetworkPolicy metrics (prometheus)
+	metrics.InitializeNetworkPolicyMetrics()
+	controller, clientset, reconciler := newTestController()
+
+	addressGroupWatcher := watch.NewFake()
+	appliedToGroupWatcher := watch.NewFake()
+	networkPolicyWatcher := watch.NewFake()
+	clientset.AddWatchReactor("addressgroups", k8stesting.DefaultWatchReactor(addressGroupWatcher, nil))
+	clientset.AddWatchReactor("appliedtogroups", k8stesting.DefaultWatchReactor(appliedToGroupWatcher, nil))
+	clientset.AddWatchReactor("networkpolicies", k8stesting.DefaultWatchReactor(networkPolicyWatcher, nil))
+
+	protocolTCP := v1beta1.ProtocolTCP
+	port := intstr.FromInt(80)
+	services := []v1beta1.Service{{Protocol: &protocolTCP, Port: &port}}
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go controller.Run(stopCh)
+
+	// Test adding policy1 with a single rule
+	policy1 := newNetworkPolicy("policy1", []string{"addressGroup1"}, []string{}, []string{"appliedToGroup1"}, services)
+	networkPolicyWatcher.Add(policy1)
+	networkPolicyWatcher.Action(watch.Bookmark, nil)
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
```

Comment:
maybe define a local function to perform this task since you repeat the same code multiple times:
```go
func ensureNoReconcilerUpdate := func() {
	select {
	case ruleID := <-reconciler.updated:
		t.Fatalf("Expected no update, got %v", ruleID)
	case <-time.After(time.Millisecond * 100):
	}
}
// ...
ensureNoReconcilerUpdate()
```

Reply:
Sure, I'll create a function.